artifact/v3/header/header_info.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <artifact/v3/header/header.hpp>
+
+#include <vector>
+
+#include <common/expected.hpp>
+#include <common/error.hpp>
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <common/json.hpp>
+
+#include <artifact/error.hpp>
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace header {
+namespace info {
+
+namespace expected = mender::common::expected;
+namespace io = mender::common::io;
+namespace error = mender::common::error;
+namespace log = mender::common::log;
+namespace json = mender::common::json;
+
+using ExpectedPayloadType = expected::expected<vector<PayloadType>, error::Error>;
+
+ExpectedPayloadType ToPayloadTypes(const json::Json &j) {
+	if (!j.IsArray()) {
+		return expected::unexpected(parser_error::MakeError(
+			parser_error::Code::ParseError, "The JSON object is not an array"));
+	}
+	vector<PayloadType> vector_elements {};
+	size_t vector_size {j.GetArraySize().value()};
```

Comment:
We could initialize the `vector_elements` to the expected size since we know it.